### DISCUSSION TITLE
Zustand (persists middleware) only save certain object into localstroage

### DISCUSSION THREAD

**Discussion by teifler** [pmndrs/zustand#887] [Q&A] [answered]
Hi fellow hackers,

currently I am coding a crypto currency tracker with zustand. I created a store and I want to use the (persists middleware) to save into the local storage. It worked just fine but now I have two API requests and I think its to much data so the website throws the following error to the console:

![image](https://user-images.githubusercontent.com/10044263/160920028-e6e144ed-9baa-4319-a5af-c688934a7e38.png)

Live version: throws the error at the crypto converter
https://capstone-project-3hr1p9uu6-teifler.vercel.app/

Here is my current store file:
https://github.com/teifler/capstone-project/blob/9d7a7f98cdae9f34ba706c8944a799be27eb55e3/src/hooks/useStore.js
https://github.com/teifler/capstone-project/pull/34

Now I tried to just save my bookmarked items into the local stroage but I dont know if thats possible.. I tried to use particularize but it throws me some errors as well..   I just want/need to save meta into the local storage, you can see it below. So my question is is it possible to use persists in my store but just save the meta.coins object? 
![image](https://user-images.githubusercontent.com/10044263/160920738-079380f7-2048-4ee8-9aa7-3eebdbc2ad75.png)

Failed
    {
      // ...
      partialize: state => ({
        meta: state.coins?.filter(coin => coin.bookmarked),
      }),
    },
    { name: 'CryptoCloud' }
  )
);

**Comment by AnatoleLucet** [comment]
Hi. I'm not sure to understand what you're trying to filter.
It looks like `state.coins` is an object, so you cannot use `.filter` on it. I also don't understand where the `coin.bookmarked` is coming from since I do not see any `bookmarked` field in store.

**Reply by teifler** [reply]
> Hi. I'm not sure to understand what you're trying to filter. It looks like `state.coins` is an object, so you cannot use `.filter` on it. I also don't understand where the `coin.bookmarked` is coming from since I do not see any `bookmarked` field in store.

Thanks for checking, the filter was just a try to get the data I need. 

First of all the bookmark will be created in meta it depends how you call the function and it will save the meta data so in my case bookmarked.  And it will only store the bookmarked coins into the meta data.

I tried to filter the the meta to get my coins which are in meta but I think a normal map would be enough because only bookmarked coins are in there. (({meta.coins[coin.id]?.bookmarked ))

meta {
coins {
bitcoin : true
}

**Comment by teifler** [comment, marked as answer]
I found partialize so you can save just certian objects instead of the whole store and it works like a charm :)

```
    {
      name: 'CryptoCloud',
      partialize: state => ({
        meta: state.meta,
      }),
    },
  )
);
```
